### PR TITLE
fix: use correct case for TL;DR and FAQ

### LINKED ISSUE #153: TLDR admonish rendering
Right now the tldr admonishment renders as `Tldr`. It would be nice if this would render as `TL;DR` instead.

### PR DESCRIPTION
**Pull request by joshka**
Fixes: #153

image

### CODE REVIEW

File: src/resolve.rs [hunk 1]
```diff
@@ -55,12 +55,18 @@ impl AdmonitionMeta {
...
+        "faq" => "FAQ".to_owned(),
+        "" => String::new(),
+        _ => {
+            // this is necessary to ensure that we deal with UTF-8 correctly rather than byte indices
+            // source: https://stackoverflow.com/a/38406885
+            let mut chars = input.chars();
+            chars.next().unwrap().to_uppercase().collect::<String>() + chars.as_str()
+        }
     }
 }
```
Comment (tommilligan) [inline src/resolve.rs]: Can `ucfirst` be left as-is? I don't see a reason to refactor it in this PR. `as_title_case` can just call it internally.

Also, you're introducing an `unwrap` here (I see that you handle the `""` case explicitly in your new code, but I think it's higher complexity for no actual performance gain).

File: src/resolve.rs [hunk 2]
```diff
@@ -55,12 +55,18 @@ impl AdmonitionMeta {
...
 /// Make the first letter of `input` upppercase.
 ///
-/// source: https://stackoverflow.com/a/38406885
-fn ucfirst(input: &str) -> String {
-    let mut chars = input.chars();
-    match chars.next() {
-        None => String::new(),
-        Some(f) => f.to_uppercase().collect::<String>() + chars.as_str(),
+/// We special case a few words to make them look nicer (e.g. "tldr" -> "TL;DR" and "faq" -> "FAQ").
+fn as_title_case(input: &str) -> String {
```
Comment (tommilligan) [inline src/resolve.rs]: nit: `as_title_case` isn't actually correct here, given the special casing.

I think renaming `ucfirst` to `as_title_case` is good to improve readability, but this new function with the special cases should be something more semantic like `directive_default_title`?

File: src/resolve.rs [hunk 3]
```diff
@@ -69,6 +75,18 @@ mod test {
...
+    fn title_case() {
+        assert_eq!(as_title_case(""), "");
+        assert_eq!(as_title_case("a"), "A");
+        assert_eq!(as_title_case("tldr"), "TL;DR");
+        assert_eq!(as_title_case("faq"), "FAQ");
+        assert_eq!(as_title_case("note"), "Note");
+        assert_eq!(as_title_case("abstract"), "Abstract");
+        // Unicode
+        assert_eq!(as_title_case("🦀"), "🦀");
+    }
```
Comment (tommilligan) [inline src/resolve.rs]: Thanks for adding tests :+1:

**Review by tommilligan — CHANGES_REQUESTED** [review]
Thanks, this looks good - just a couple of nits for future maintainability, but nothing major

File: src/resolve.rs (hunk 1 above)
Comment (joshka) [reply src/resolve.rs]: Done in https://github.com/tommilligan/mdbook-admonish/pull/154/commits/fd7b04545f8b232ac3bb0ea5ae40dc0f095eca39

File: src/resolve.rs (hunk 2 above)
Comment (joshka) [reply src/resolve.rs]: Title case would be every first letter of each word uppercase. I called it format_directive_title in https://github.com/tommilligan/mdbook-admonish/pull/154/commits/d5bdde1f5cee6d3a8331eba82dcdc4cc8e5a02ea

**Review by tommilligan — APPROVED** [review]
Amazing, thanks!

**Pull request merged**